server.js

```diff
@@ -258,20 +258,12 @@ module.exports.initExpress = function () {
     logLevel: 'silent',
     logProvider: (_provider) => logger,
     router: async (req) => {
-      try {
-        const match = req.url.match(/^\/pl\/workspace\/([0-9]+)\/container\//);
-        if (!match) throw new Error(`Could not match URL: ${req.url}`);
-        const workspace_id = match[1];
-        const result = await sqldb.queryOneRowAsync(
-          `SELECT hostname FROM workspaces WHERE id = $workspace_id;`,
-          { workspace_id }
-        );
-        const url = `http://${result.rows[0].hostname}/`;
-        return url;
-      } catch (err) {
-        logger.error(`Error in router for url=${req.url}: ${err}`);
-        return 'not-matched';
+      if (!req.workspace_hostname) {
```

Comment:
I removed the `try`/`catch` around all this that returned `not-matched` if an error was encountered. AFAICT, that wasn't actually treated as a special value by any of the `http-proxy` code; I'd rather just throw an error than try to route to a nonsensical hostname.

Reply:
🙏 i agree this seems like a much more sensible thing to do